services/frontend-service/src/ui/Batch.test.tsx

```diff
@@ -188,75 +235,67 @@ describe('Confirmation Dialog Provider', () => {
-        {
-            action: {
-                $case: 'createEnvironmentLock',
-                createEnvironmentLock: {
-                    environment: 'dummy environment',
-                    lockId: '1234',
-                    message: 'hello',
-                },
-            },
-        },
-        {
-            action: {
-                $case: 'createEnvironmentApplicationLock',
-                createEnvironmentApplicationLock: {
-                    environment: 'dummy environment',
-                    application: 'dummy application',
-                    lockId: '1234',
-                    message: 'hello',
-                },
-            },
-        },
-    ];
+    const sampleCartActions: BatchAction[] = [newVersionDeploy, newIdEnvLock, newIdAppLock];
 
     describe.each(data)(`Batch Action Types`, (testcase: dataT) => {
         it(`${testcase.type}`, () => {
             // given
-            const { container } = getWrapper({ action: testcase.act, fin: testcase.fin, locks: testcase.locks });
+            const { container } = getWrapper(
+                { action: testcase.act, fin: testcase.fin, locks: testcase.locks },
+                sampleCartActions
+            );
 
-            // when - open the confirmation dialog
-            expect(container.querySelector('#dialog-opener')!.textContent).toBe('ClickMe');
-            fireEvent.click(container.querySelector('#dialog-opener')!);
+            // when - clicking the button
+            expect(container.querySelector('#dummy-button')!.textContent).toBe('ClickMe');
+            fireEvent.click(container.querySelector('#dummy-button')!);
 
+            // test for conflicts
+            // when adding the same action should be flagged as conflict
+            const res = exportedForTesting.isConflictingAction(sampleCartActions, testcase.act);
             // then
-            const title = document.querySelector('.confirmation-title');
-            expect(title!.textContent).toBe(testcase.expect.title);
+            expect(res).toBe(testcase.expect.conflict);
 
-            // when - clicking yes
-            const d = document.querySelector('.MuiDialog-root');
-            fireEvent.click(getByText(d! as HTMLElement, 'Add to cart').closest('button')!);
+            if (testcase.expect.conflict) {
+                // then a dialog shows up
+                const title = document.querySelector('.confirmation-title');
+                expect(title!.textContent).toBe(testcase.expect.title);
 
-            // then
-            mock_setActions.wasCalledWith([testcase.act], Spy.IGNORE);
+                // when - clicking yes
+                const d = document.querySelector('.MuiDialog-root');
+                fireEvent.click(getByText(d! as HTMLElement, 'Add anyway').closest('button')!);
 
+                // then
+                mock_setActions.wasCalledWith([...sampleCartActions, testcase.act], Spy.IGNORE);
+
+                if (testcase.locks) {
```

Comment:
why is this `if (testcase.locks)` nested within the other if `if (testcase.expect.conflict)`?
Whether or not you expect locks should be independent of whether you expect a conflict.
-> Please move this to top level, if possible

Reply:
(I'm trying to remove nested if statements in tests, to keep it simple)